code/controllers/subsystem/discord.dm

```diff
@@ -104,8 +104,8 @@ SUBSYSTEM_DEF(discord)
 	qdel(query_get_discord_ckey)
 
 // Finalises link
-/datum/controller/subsystem/discord/proc/link_account(ckey)
-	var/datum/DBQuery/link_account = SSdbcore.NewQuery("UPDATE [format_table_name("player")] SET discord_id = '[sanitizeSQL(account_link_cache[ckey])]' WHERE ckey = '[sanitizeSQL(ckey)]'")
+/datum/controller/subsystem/discord/proc/link_account(ckey, discordid)
+	var/datum/DBQuery/link_account = SSdbcore.NewQuery("UPDATE [format_table_name("player")] SET discord_id = '[sanitizeSQL(discordid)]' WHERE ckey = '[sanitizeSQL(ckey)]'")

```

Comment:
Table uses BIGINT and sanitizeSQL turns it into a quoted string. Unsure if this will work, but it probably will.

Reply:
this is basically what it was doing before though, just the lookup was moved explicitly up a function

Reply:
We don't use strict mode so it'll convert to an int automatically.

Reply:
The 'proper' way is to run it through text2num, which will ensure a text string is converted to a number or otherwise return the number. 

This works thou